core/templates/span.h

```diff
@@ -62,4 +62,54 @@ class Span {
 
 	_FORCE_INLINE_ constexpr const T *begin() const { return _ptr; }
 	_FORCE_INLINE_ constexpr const T *end() const { return _ptr + _len; }
+
+	// Algorithms.
+	int64_t find(const T &p_val, int64_t p_from = 0) const;
+	int64_t rfind(const T &p_val, int64_t p_from = 0) const;
+	uint64_t count(const T &p_val) const;
```

Comment:
Can safely prepend `constexpr` to these.
```suggestion
	constexpr int64_t find(const T &p_val, int64_t p_from = 0) const;
	constexpr int64_t rfind(const T &p_val, int64_t p_from = 0) const;
	constexpr uint64_t count(const T &p_val) const;
```

Reply:
Sidenote: all instances of `_FORCE_INLINE_` are redundant here, as `constexpr` implies `inline` for functions.

Reply:
> Can safely prepend constexpr to these.

You're totally right!

> Sidenote: all instances of _FORCE_INLINE_ are redundant here, as constexpr implies inline for functions.

Technically, just being declared as a member function already implies inline, at least if CLion is right:

<img width="817" alt="SCR-20250316-dmff" src="https://github.com/user-attachments/assets/41ca7d9e-12f9-4c0b-a7d2-ce82f736730b" />

`FORCE_INLINE` has an additional effect because it will (likely) inline the functions even on `-O0`, when this is not at all guaranteed by the `inline` keyword.
Somebody said they tested compiling `Godot` without all the `FORCE_INLINE` keywords, and noticed substantial performance decreases for dev builds. So they do seem to help somewhat.

Reply:
I'm not sure if that applies in contexts with `constexpr`, but I suppose it doesn't hurt. Might as well prepend `_FORCE_INLINE_` to these 3 functions too

Reply:
I don't think I want to force the compiler to force inlining the functions. If it saves significant space, it might be wiser to allow it to generate actual functions for them instead.
That's why I didn't mark these 3 as `FORCE_INLINE`.